File: Day 2/problem2/D_Destruction_of_the_Dandelion_Fields.cpp
                                      /*بسم الله الرحمن الرحيم*/       
#include <algorithm>
#include <iostream>
#include <numeric>
#include <vector>
using namespace std;
#define Abd_ElQader_ ios_base::sync_with_stdio(false),cin.tie(NULL),cout.tie(NULL);
#define len(v) (int)v.size()
#define Ceil(a,b) ((ll)a+(ll)(b-1))/b
typedef long long ll;

void solve() {
    int n;
    cin >> n;
    vector<int>v(n);
    int odd_s=0,even_s=0;
    ll ans = 0;
    for (int i = 0; i < n; i++) {
        cin >> v[i];
        if(v[i]&1){odd_s++;}
        else {even_s++;}
    }
    if(!odd_s) {
        cout << "0\n";
        return;
    }

    ans = accumulate(v.begin(), v.end(), 0ll);
    sort(v.begin(), v.end());
    odd_s = (odd_s)/2;
    for (int i = 0; i < n && odd_s; i++) {
        if(v[i]&1) {
            odd_s--;
            ans -= v[i];
        }
    }

    cout << ans << endl;
}  

signed main() {
    Abd_ElQader_

    int t=1;
    cin>>t;
    while(t--) 
    solve();
}

File: Day 2/problem2/D_Destruction_of_the_Dandelion_Fields.py
def solve():
    n = int(input())
    l = [int(x) for x in input().split(" ")]
    odd_s = len([x for x in l if x&1])
    evens = len([x for x in l if x&1^1])

    if not odd_s:
        print(0)
        return

    ans = sum(l)
    l = sorted(l)
    odd_s = int(odd_s/2) 
    i = 0

    while odd_s and i < n:
        if l[i]&1:
            odd_s -= 1
            ans -= l[i]
        i += 1

    print(ans)

t = int(input())
while t:
    solve()
    t -= 1
